app/controllers/assessment360_controller.rb

```diff
@@ -90,48 +94,51 @@ def course_student_grade_summary
-      flash[:error] = "There is no course participant in course #{course.name}"
-      redirect_to(:back)
-    end
-
+    inspect_course_participants(@course_participants)
+    
     @course_participants.each do |cp|
       @topics[cp.id] = {}
       @assignment_grades[cp.id] = {}
       @peer_review_scores[cp.id] = {}
       @final_grades[cp.id] = 0
-
       @assignments.each do |assignment|
         user_id = cp.user_id
         assignment_id = assignment.id
-
         assignment_participant = assignment.participants.find_by(user_id: user_id)
         next if assignment.participants.find_by(user_id: user_id).nil?
+        next if TeamsUser.team_id(assignment_id, user_id).nil?
+        assignment_grade_summary(cp,assignment_id, user_id)
 
-        # A topic exists if a team signed up for a topic, which can be found via the user and the assignment
-        topic_id = SignedUpTeam.topic_id(assignment_id, user_id)
-        @topics[cp.id][assignment_id] = SignUpTopic.find_by(id: topic_id)
-
-        # Instructor grade is stored in the team model, which is found by finding the user's team for the assignment
-        team_id = TeamsUser.team_id(assignment_id, user_id)
-        next if team_id.nil?
-
-        team = Team.find(team_id)
         peer_review_score = find_peer_review_score(user_id, assignment_id)
-
-        # Set the assignment grade, peer review score, and sum for the final student summary
-        @assignment_grades[cp.id][assignment_id] = team[:grade_for_submission]
-        unless @assignment_grades[cp.id][assignment_id].nil?
-          @final_grades[cp.id] += @assignment_grades[cp.id][assignment_id]
-        end
-
-        unless (peer_review_score.nil? || peer_review_score[:review][:scores][:avg].nil?)
+        unless peer_review_score.dig(:review,:scores,:avg).nil?
           @peer_review_scores[cp.id][assignment_id] = peer_review_score[:review][:scores][:avg].round(2)
         end
       end
     end
   end
 
+
+
+  def assignment_grade_summary(cp,assignment_id, user_id)
+    # A topic exists if a team signed up for a topic, which can be found via the user and the assignment
+    topic_id = SignedUpTeam.topic_id(assignment_id, user_id)
+    @topics[cp.id][assignment_id] = SignUpTopic.find_by(id: topic_id)
+        # Instructor grade is stored in the team model, which is found by finding the user's team for the assignment
+    team_id = TeamsUser.team_id(assignment_id, user_id)
+    team = Team.find(team_id)
+    @assignment_grades[cp.id][assignment_id] = team[:grade_for_submission]
+    unless @assignment_grades[cp.id][assignment_id].nil?
+      @final_grades[cp.id] += @assignment_grades[cp.id][assignment_id]
+    end
+  end
+
+  def inspect_course_participants(course_participants)
+    if course_participants.empty?
+      flash[:error] = "There is no course participant in course #{course.name}"
+      redirect_to(:back)
+    end
+  end
+
   def populate_hash_for_all_students_all_reviews(assignment,
```

Comment:
This is a bad name too, because "populate_hash" doesn't give any hint of what this is computing, except that the result will be some hash.